src/main/java/com/aws/greengrass/deployment/IotJobsHelper.java

```diff
@@ -268,17 +276,45 @@ private static void unwrapExecutionException(ExecutionException e)
     }
 
     @Override
-    @SuppressFBWarnings
+    @SuppressWarnings("PMD.CollapsibleIfStatements")
     public void postInject() {
+        deviceConfiguration.onAnyChange((what, node) -> {
+            if (node != null &&  WhatHappened.changed.equals(what)) {
+                // List of configuration nodes that we need to reconfigure for if they change
+                if (node.childOf(DEVICE_MQTT_NAMESPACE) || node.childOf(DEVICE_PARAM_THING_NAME)
```

Comment:
you should ignore mqtt and auth changes. The mqtt client handles that already. Only listen to things that affect what job topics we subscribe to.

Reply:
ok